File: pubsub/src/main/java/glitch/pubsub/events/StreamDownEvent.java
package glitch.pubsub.events;

import glitch.api.ws.events.AbstractEvent;
import glitch.api.ws.events.IEvent;
import glitch.pubsub.GlitchPubSub;
import glitch.pubsub.Topic;
import lombok.Data;
import lombok.EqualsAndHashCode;

import java.time.Instant;

@Data
@EqualsAndHashCode(callSuper = true)
public class StreamDownEvent extends AbstractEvent<GlitchPubSub> implements IEvent<GlitchPubSub> {
    private final Topic topic;
    private final Instant serverTime;
    public StreamDownEvent(GlitchPubSub client, Topic topic, Instant serverTime) {
        super(client);
        this.topic = topic;
        this.serverTime = serverTime;
    }
}

File: pubsub/src/main/kotlin/glitch/pubsub/events/StreamDownEvent.kt
package glitch.pubsub.events

import glitch.api.ws.events.AbstractEvent
import glitch.api.ws.events.IEvent
import glitch.pubsub.GlitchPubSub
import glitch.pubsub.Topic
import java.time.Instant

class StreamDownEvent(client: GlitchPubSub, val topic: Topic, val serverTime: Instant) : AbstractEvent<GlitchPubSub>(client), IEvent<GlitchPubSub> {
}
